src/DataProvider/ChainCollectionDataProvider.php

```diff
@@ -39,8 +40,14 @@ public function getCollection(string $resourceClass, string $operationName = nul
     {
         foreach ($this->dataProviders as $dataProvider) {
             try {
+                if ($dataProvider instanceof RestrictedDataProviderInterface
+                    && !$dataProvider->supports($resourceClass, $operationName)) {
+                    continue;
+                }
+
                 return $dataProvider->getCollection($resourceClass, $operationName);
             } catch (ResourceClassNotSupportedException $e) {
+                @trigger_error(sprintf('Throwing a "%s" is deprecated in favor of implementing "%s"', get_class($e), RestrictedDataProviderInterface::class), E_USER_DEPRECATED);
```

Comment:
Following the sf style here... yuck!

Reply:
Why not `ResourceClassNotSupportedException::class` constant? Also, could you mention "Throwing a "..." **in a data provider** is deprecated..."?

Reply:
Can you just make this change and we'll merge.

Reply:
Oh sure I forgot. Let's see what those constants are

Reply:
Oh right I understand now that I see the code in my editor, I just put the wrong class constant. I could have understood this earlier had it been legible on github :P
Fun fact: this line does not even fit on a 1920x1200 display.

Reply:
Fixed @dunglas